src/run/runner/executor.rs

```diff
@@ -32,4 +34,20 @@ pub trait Executor {
         system_info: &SystemInfo,
         run_data: &RunData,
     ) -> Result<()>;
+
+    /// Gets the base environment for the command
+    ///
+    /// Later on, we will want to refactor this and create the cmd directly in a trait function
+    fn get_cmd_base_envs(&self, profile_folder: &Path) -> HashMap<&str, String> {
+        let mut envs = HashMap::new();
+        envs.insert("CODSPEED_RUNNER_MODE", self.name().to_string());
+        envs.insert(
+            "CODSPEED_PROFILE_FOLDER",
+            profile_folder.to_str().unwrap().to_string(),
+        );
+        for (key, value) in BASE_INJECTED_ENV.iter() {
+            envs.insert(key, value.to_string());
+        }
+        envs
```

Comment:
we can create the hashmap directly from BASE_INJECTED_ENV, either cloning or with `HashMap::from`